File: ployglots/mta-ployglot-pkg-vault/packages/py/src/polyglot_vault_file/logger.py
"""Logger + LogLevel."""
from __future__ import annotations

import logging
from enum import IntEnum
from typing import Protocol, Any


class LogLevel(IntEnum):
    DEBUG = 0
    INFO = 1
    WARN = 2
    ERROR = 3
    NONE = 4


class IVaultFileLogger(Protocol):
    def debug(self, msg: str, *args: Any) -> None: ...
    def info(self, msg: str, *args: Any) -> None: ...
    def warn(self, msg: str, *args: Any) -> None: ...
    def error(self, msg: str, *args: Any) -> None: ...


class Logger:
    def __init__(self, package_name: str, filename: str) -> None:
        self._pkg = package_name
        self._file = filename
        self._log = logging.getLogger(f"{package_name}.{filename}")
        if not self._log.handlers:
            handler = logging.StreamHandler()
            handler.setFormatter(logging.Formatter("%(message)s"))
            self._log.addHandler(handler)

    @staticmethod
    def create(package_name: str, filename: str) -> "IVaultFileLogger":
        return Logger(package_name, filename)  # type: ignore[return-value]

    def _format(self, message: str) -> str:
        return f"[{self._pkg}][{self._file}] {message}"

    def debug(self, msg: str, *args: Any) -> None:
        if _current_log_level <= LogLevel.DEBUG:
            self._log.debug(self._format(msg), *args)

    def info(self, msg: str, *args: Any) -> None:
        if _current_log_level <= LogLevel.INFO:
            self._log.info(self._format(msg), *args)

    def warn(self, msg: str, *args: Any) -> None:
        if _current_log_level <= LogLevel.WARN:
            self._log.warning(self._format(msg), *args)

    def error(self, msg: str, *args: Any) -> None:
        if _current_log_level <= LogLevel.ERROR:
            self._log.error(self._format(msg), *args)


_current_log_level: LogLevel = LogLevel.DEBUG


def set_log_level(level: LogLevel) -> None:
    global _current_log_level
    _current_log_level = level
    logging.getLogger().setLevel({
        LogLevel.DEBUG: logging.DEBUG,
        LogLevel.INFO: logging.INFO,
        LogLevel.WARN: logging.WARNING,
        LogLevel.ERROR: logging.ERROR,
        LogLevel.NONE: logging.CRITICAL + 1,
    }[level])


_default_logger: IVaultFileLogger = Logger.create("polyglot_vault_file", "default")


def get_logger() -> IVaultFileLogger:
    return _default_logger

File: ployglots/mta-ployglot-pkg-vault/packages/ts/src/logger.ts
// @ts-nocheck
export enum LogLevel {
  DEBUG = 0,
  INFO = 1,
  WARN = 2,
  ERROR = 3,
  NONE = 4,
}

let currentLevel: LogLevel = LogLevel.DEBUG;

export interface IVaultFileLogger {
  debug(msg: string, ...args: unknown[]): void;
  info(msg: string, ...args: unknown[]): void;
  warn(msg: string, ...args: unknown[]): void;
  error(msg: string, ...args: unknown[]): void;
}

export class Logger implements IVaultFileLogger {
  private constructor(
    private readonly packageName: string,
    private readonly fileName: string,
  ) {}

  static create(packageName: string, fileName: string): IVaultFileLogger {
    return new Logger(packageName, fileName);
  }

  private prefix(): string {
    return `[${this.packageName}][${this.fileName}]`;
  }

  debug(msg: string, ...args: unknown[]): void {
    if (currentLevel <= LogLevel.DEBUG) console.debug(this.prefix(), msg, ...args);
  }
  info(msg: string, ...args: unknown[]): void {
    if (currentLevel <= LogLevel.INFO) console.info(this.prefix(), msg, ...args);
  }
  warn(msg: string, ...args: unknown[]): void {
    if (currentLevel <= LogLevel.WARN) console.warn(this.prefix(), msg, ...args);
  }
  error(msg: string, ...args: unknown[]): void {
    if (currentLevel <= LogLevel.ERROR) console.error(this.prefix(), msg, ...args);
  }
}

const defaultLogger: IVaultFileLogger = Logger.create('vault-file', 'default');

export function getLogger(): IVaultFileLogger {
  return defaultLogger;
}

export function setLogLevel(level: LogLevel): void {
  currentLevel = level;
}
